### ISSUE TITLE
guide: GH resuming workflow

### ISSUE THREAD

**Issue by casperdcl** [iterative/cml.dev#207]
Add a self-hosted long-running example to https://cml.dev/doc/cml-with-dvc (or somewhere else)

1. GH action launches "self-hosted" GCP/AWS using `cml runner --reuse --labels=cml`
2. GH actions runs the rest of the workflow on the "self-hosted" runner using `runs-on: [self-hosted, cml]`
3. If GH actions is about to timeout, CML will restart the workflow

- i.e. https://cml.dev/doc/self-hosted-runners?tab=GitHub#allocating-cloud-compute-resources-with-cml
- The key is requesting GH's maximum `timeout-minutes: 4320- - this signals to CML to restart the workflow just before timeout.
- write code to cache results so that the restarted workflow will use previous results (e.g. use https://dvc.org/doc/user-guide/experiment-management/checkpoints#caching-checkpoints)

**Comment by casperdcl** [comment]
more musings (for `cml runner --cloud-spot`):

```python
live = dvclive.Live(resume=True)
model = Model(load="model.pkl" if Path("model.pkl").exists() else None)
while (epoch := live.get_step()) < 100:
    history = model.fit(X, Y, epochs=1)
    if epoch % 10 == 0:  # at most 10 epochs are lost upon CML respawing a spot instance
        model.save("model.pkl")
    live.log("loss", history['loss'])
    live.next_step()
```

**Comment by jorgeorpinel** [comment]
Out of curiosity, what makes this p1? Perhaps there are there lots of support cases that could be avoided by or redirected to this? Thanks

**Comment by casperdcl** [comment]
lots of support requests over YEARS; super overdue.

**Comment by omesser** [comment]
deprioritized and frozen. Removing from CML project board for now
